datafusion/physical-plan/src/projection.rs

```diff
@@ -74,11 +73,14 @@ pub struct ProjectionExec {
 
 impl ProjectionExec {
     /// Create a projection on an input
-    pub fn try_new(
-        expr: Vec<ProjectionExpr>,
-        input: Arc<dyn ExecutionPlan>,
-    ) -> Result<Self> {
+    pub fn try_new<I, E>(expr: I, input: Arc<dyn ExecutionPlan>) -> Result<Self>
```

Comment:
I made this generic so it could take both the old and new structures

It would be really nice to add an example code here too